### ISSUE TITLE
Check whether Layer 7 protocol docs work before v1.17 release

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#36173]
~Some Layer 7 network policies are beta and require manually building Envoy with Go extensions, but the documentation doesn't make this very clear. For example,~ The [Cassandra](https://docs.cilium.io/en/stable/security/cassandra/) guide implies that you can just install a standard quick-install version of Cilium to get up and running, but this does not work. Not only this, but there is no validation in the network policy engine to inform the user that the deployed software will not implement their requested policy.

In practice when I ran through the Cassandra guide with a relatively default configuration (via local kind dev environment including external Envoy), I found that (a) it requires 10GB+ of memory (surprise!) and (b) once the policy is applied, all connections to the cassandra server are broken. This makes the guide effectively broken if you just follow the guide and don't know what you're doing to build & deploy custom Envoy extensions for the protocol support.

```
$ kubectl exec -it $HQ_POD -- cqlsh cassandra-svc
Connection error: ('Unable to connect to any servers', {'10.244.1.133:9042': ConnectionShutdown('Connection to 10.244.1.133:9042 was closed')})
command terminated with exit code 1
```

**Comment by joestringer** [comment]
To poke around and confirm my suspicions here, I also tried to follow the [memcached example](https://docs.cilium.io/en/stable/security/memcached/). Some of the steps didn't work, but in the end when the network policy was applied I was able to get some protocol-specific `access denied` messages with v1.17.0-pre.2. Probably for v1.17 final we should try out the various L7 protocol examples and make a call feature by feature whether to fix / update docs / drop the instructions.
